src/core_ocean/mode_init/mpas_ocn_init_tidal_boundary.F

```diff
@@ -222,32 +232,153 @@ subroutine ocn_init_setup_tidal_boundary(domain, iErr)!{{{
+              restingThickness(k, iCell) = bottomDepth(iCell)/maxLevelCell(iCell)
+            end do
+          end do
+        else
+          do iCell = 1, nCellsSolve
+            ! Set layerThickness and restingThickness
+            call ocn_compute_layerThickness_zMid_from_bottomDepth(layerThickness(:,iCell), zMid(:,iCell), &
+              refBottomDepth, bottomDepth(iCell), maxLevelCell(iCell), nVertLevels, iErr, &
+              restingThickness=restingThickness(:,iCell), &
+              ssh=ssh(iCell))
+            ! also computes zMid
+            do k = 1, maxLevelCell(iCell)
+              restingThickness(k, iCell) = layerThickness(k, iCell)
+            end do
           end do
         end if
 
-      end do
+      else if (trim(config_tidal_boundary_layer_type) == 'sigma') then
+
+        do iCell = 1, nCellsSolve
+          maxLevelCell(iCell) = nVertLevels
+        end do
+
+        if (config_use_wetting_drying .and. config_tidal_start_dry) then
+          do iCell = 1, nCellsSolve
+            ssh(iCell) = -bottomDepth(iCell) + config_drying_min_cell_height*maxLevelCell(iCell)
+            ! also computes zMid
+            do k = 1, maxLevelCell(iCell)
+              layerThickness(k,iCell) = (ssh(iCell) + bottomDepth(iCell))/maxLevelCell(iCell)
+              restingThickness(k, iCell) = bottomDepth(iCell)/maxLevelCell(iCell)
+            end do
+          end do
+        else
+          do iCell = 1, nCellsSolve
+            ! Set layerThickness and restingThickness
+            call ocn_compute_layerThickness_zMid_from_bottomDepth(layerThickness(:,iCell), zMid(:,iCell), &
+              refBottomDepth, bottomDepth(iCell), maxLevelCell(iCell), nVertLevels, iErr, &
+              restingThickness=restingThickness(:,iCell), &
+              ssh=ssh(iCell))
+            ! also computes zMid
+            do k = 1, maxLevelCell(iCell)
+              restingThickness(k, iCell) = layerThickness(k, iCell)
+            end do
+          end do
+        end if
+
+      else if (trim(config_tidal_boundary_layer_type) == 'hybird') then
+
+        do iCell = 1, nCellsSolve
+          maxLevelCell(iCell) = nVertLevels
+          if (nVertLevels > 1) then
+            do k = 1, nVertLevels
+              if (bottomDepth(iCell) < refBottomDepth(k)) then
+                if (k < config_tidal_boundary_min_vert_levels) then  
+                  ! Use sigma
+                  maxLevelCell(iCell) = config_tidal_boundary_min_vert_levels
+                  exit
+                else
+                  ! Use zstar
+                  maxLevelCell(iCell) = k-1
+                  bottomDepth(iCell) = refBottomDepth(k-1)
+                  exit
+                end if
+              end if
+            end do
+          end if
+
+          ssh(iCell) = 0.0_RKIND
+        end do
+
+        if (config_use_wetting_drying .and. config_tidal_start_dry) then                                     ! this is all repeated code
```

Comment:
@pwolfram, this section of code seems to be the same across the zstar, sigma, and hybird cases. I could be missing something, but if that's the case, it should be taken out of the if statements for each case and moved outside.

Reply:
Agreed, thanks!